src/containers/Tags.jsx

```diff
@@ -0,0 +1,280 @@
+import React from "react";
+import { Card, CardText } from "material-ui/Card";
+import Chip from "../components/Chip";
+import CreateIcon from "material-ui/svg-icons/content/create";
+import DeleteIcon from "material-ui/svg-icons/action/delete-forever";
+import RaisedButton from "material-ui/RaisedButton";
+import { red500 } from "material-ui/styles/colors";
+import FloatingActionButton from "material-ui/FloatingActionButton";
+import ContentAdd from "material-ui/svg-icons/content/add";
+import theme from "../styles/theme";
+import Dialog from "material-ui/Dialog";
+import yup from "yup";
+import GSForm from "../components/forms/GSForm";
+import Form from "react-formal";
+import { StyleSheet, css } from "aphrodite";
+import loadData from "./hoc/load-data";
+import gql from "graphql-tag";
+import { withRouter } from "react-router";
+import PropTypes from "prop-types";
+
+const styles = StyleSheet.create({
+  cards: {
+    display: "flex",
+    flexWrap: "wrap"
+  },
+  card: {
+    marginRight: 10,
+    marginBottom: 10
+  },
+  buttons: {
+    display: "flex"
+  },
+  editButton: {
+    marginRight: 6
+  },
+  fields: {
+    display: "flex",
+    flexDirection: "column",
+    marginBotton: 16
+  },
+  submit: {
+    marginTop: 32,
+    display: "flex",
+    justifyContent: "flex-end"
+  }
+});
+
+export class Tags extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      openTagDialog: false,
+      tagId: null,
+      dialogTitle: "",
+      dialogSubmitHandler: null,
+      dialogButtonLabel: ""
+    };
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpenEdit = this.handleOpenEdit.bind(this);
+    this.handleCreate = this.handleCreate.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  handleOpen() {
+    this.setState({
+      openTagDialog: true,
+      dialogTitle: "Create Tag",
+      dialogSubmitHandler: this.handleCreate,
+      dialogButtonLabel: "Create",
+      tagId: null
+    });
+  }
+
+  handleOpenEdit(tagId) {
+    return () =>
+      this.setState({
+        openTagDialog: true,
+        dialogTitle: "Edit Tag",
+        dialogSubmitHandler: this.handleEdit,
+        dialogButtonLabel: "Save",
+        tagId
+      });
+  }
+
+  handleClose() {
+    this.setState({ openTagDialog: false });
+  }
+
+  async handleCreate(formData) {
+    await this.props.mutations.createTag(
+      formData,
+      this.props.params.organizationId
+    );
+    await this.props.data.refetch();
+    this.handleClose();
+  }
+
+  async handleEdit(formData) {
+    await this.props.mutations.editTag(
+      formData,
+      this.props.params.organizationId,
+      this.state.tagId
+    );
+    await this.props.data.refetch();
```

Comment:
@matteosb I added this line in all of the mutation event handlers to refetch data after the mutation. Without it, dom updates don't seem to work. But if I made the changes you suggested above for updating the cache correctly after a mutation, should a refetch be necessary? My experience with GraphQL is limited, so I may be missing something. Thanks

Reply:
Interesting, this shouldn't be necessary if you are returning the fields you are updating from the mutation but we are currently on an ancient version of apollo-client, so it might be broken. It's fine to leave this in here for now. I'm going to do a pass on this once I upgrade apollo-client (see the discussion here: #1461)

Reply:
@matteosb Yeah, I noticed there isn't an apollo-cache package installed, and I wasn't totally clear on the caching strategies being used overall. I guess we'll see what can be changed with the upgrade. Thanks!